### PR TITLE
Add nix flake support

### PR DESCRIPTION
**Pull request by emctoo**

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            "clippy"
+            "rust-src"
+          ];
+
+          craneLib = (crane.mkLib pkgs).overrideToolchain rustToolchain;
+
+          commonArgs = {
+            src = craneLib.cleanCargoSource ./.;
+            pname = "rs1090";
+            version = "v0.2.2";
```
[the remark this answers is not in the document]
Comment (emctoo) [reply flake.nix:47]: I will try to resolve this in my PR #18. Close this for now.

[the remark this answers is not in the document]
Comment (emctoo) [reply flake.nix]: This is resolved.

File: justfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
+default:
+    @just --list
+
+# Auto-format the source tree
+fmt:
+    treefmt
```
Comment (xoolive) [inline justfile:6]: what is treefmt?

File: rust-toolchain.toml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+[toolchain]
+channel = "stable"
```
Comment (xoolive) [inline rust-toolchain.toml:2]: is this file really necessary?

File: .gitignore [hunk 4]
```diff
@@ -1,3 +1,4 @@
 /target
 
 basestation.*
+.direnv/
```
Comment (xoolive) [inline .gitignore:4]: is this really necessary? I would rather see it in one's global .gitignore, no?

Comment (emctoo) [reply .gitignore:4]: Makes sense. I should put it in my global configuration. Thanks.

File: rust-toolchain.toml (hunk 3 above)
Comment (emctoo) [reply rust-toolchain.toml:2]: I'm not sure. For a time, this library only compiles with 1.76 tool chain. Do we have requirements for compatibility?

File: justfile (hunk 2 above)
Comment (emctoo) [reply justfile:6]: A command line tool to rule all formatting, [treefmt](https://github.com/numtide/treefmt)

Just a preference. We should stick to Rust/Python ecosystem tools, better accepted for the community.

**Comment by xoolive** [comment]
Hi @emctoo do you think we should try to clean this PR and try to merge it?
